tests/virt_autotest/qem/live_migration_src.pm

```diff
@@ -0,0 +1,213 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Live migration test for source host
+#
+# - Get resource data and dst host IP address
+# - Collect guest data, name, version, arch, image path, reusability ...
+# - Generate profile from autoyast template based on guest data
+# - First stage guest install
+# - Second stage guest install
+# - Get IP address of Guest and web page content. installation complete
+# - Set up passwordless ssh authentication
+# - Execute migration test cases
+# - Heath check guest on the dts host after migration
+# - clean up
+#
+# Maintainer: Tony Yuan <tyuan@suse.com>
+
+package live_migration_src;
+use strict;
+use warnings;
+use base 'opensusebasetest';
+use testapi;
+use File::Copy 'copy';
+use File::Path 'make_path';
+use lockapi;
+use mmapi;
+use scheduler 'get_test_suite_data';
+
+my $vms            = {};
+my $vms_to_install = 0;
+my $images_path    = "/var/lib/libvirt/images";
+my $nfs_path       = get_var("NFS_PATH") ? get_var("NFS_PATH") : get_var('SUT_IP') . ":$images_path";
+my $guest          = get_required_var('GUEST');
+my $virt_insatll = "/usr/bin/virt-install --noautoconsole --network bridge=br0 --vcpus 2,maxvcpus=4 --memory 2048,maxmemory=4096 --events on_reboot=restart --video=vga --serial pty -x 'quiet textmode=1 sshd=1 sshpassword=novell'";
+my $sut_ip = get_var('SUT_IP');
+#my $inst_src = get_test_suite_data()->{location}->{"$guest"};
+my $inst_src;
+my $dst_ip;
+
+#collect parameters for each vm.
+sub init_vm_info {
+    my $ret;
+    if (check_var('SYSTEM_ROLE', 'kvm')) {
+        $ret = script_run("[ -f $images_path/vm-configs/$guest-kvm.xml ]");
+        if ($ret == 0) {
+            assert_script_run("virsh define --file $images_path/vm-configs/$guest-kvm.xml");
+            $vms->{"$guest-kvm"} = {reuse => 1};
+        } else {
+            $vms->{"$guest-kvm"} = {
+                version   => $guest =~ /-sp/ ? $guest =~ s/.*-(\d+)-sp(\d)/$1.$2/r : $guest =~ s/.*-(\d+)/$1/r,
+                arch      => get_var('ARCH'),
+                location  => $inst_src,
+                path      => "$images_path/$guest-kvm.qcow2,size=20",
+                port      => 5906,
+                disk_type => "vda"
+            };
+            $vms_to_install++;
+        }
+        $vms->{"$guest-kvm"}->{dsturi}    = "qemu+ssh://$dst_ip/system";
+        $vms->{"$guest-kvm"}->{virsh_cmd} = "virsh migrate --live --p2p --change-protection --unsafe --compressed --abort-on-error";
```

Comment:
Why are the `--live --p2p ...` settings set in the `virsh migrate` command and not below in the `migrate_opts`?

Reply:
It will migrate guest with each different opts. I put the different opts in migrate_opts. The opt in migrate_opts will be looped to construct different migrate commands.